parser/encoding/request.go

```diff
@@ -57,13 +84,103 @@ type ParameterEncoding struct {
 	// The location specific name of the parameter (e.g. cheeseEater, cheese-eater, X-Cheese-Eater
 	Name string
 	// The underlying field definition
-	Field *v1.Field
+	Field *schema.Field
 }
 
-// DescribeRequest combines a payload v1.Struct and a list of HTTP methods and returns
+// DescribeRPC expresses how to encode an RPCs request and response objects for the wire.
+func DescribeRPC(appMetaData *meta.Data, rpc *meta.RPC) (*RPCEncoding, error) {
+	encoding := &RPCEncoding{}
+	// Work out the request encoding
+	requestStruct, err := getConcreteStructType(appMetaData, rpc.RequestSchema, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "request struct")
+	}
+	encoding.RequestEncoding, err = DescribeRequest(requestStruct, rpc.HttpMethods...)
+	if err != nil {
+		return nil, errors.Wrap(err, "request encoding")
+	}
+
+	// Work out the response encoding
+	_, err = getConcreteStructType(appMetaData, rpc.ResponseSchema, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "response struct")
+	}
+	// TODO(stefan): implement response encoding here
```

Comment:
@ekerfelt I've left this TODO for you to wire in your new `DescribeResponse` function from your branch.